packages/arb-validator/rollupmanager/manager.go

```diff
@@ -128,15 +128,23 @@ func CreateManagerAdvanced(
 			}
 
 			if checkpointer.HasCheckpointedState() {
-				chainObserverBytes, restoreCtx, err := checkpointer.RestoreLatestState(runCtx, clnt, rollupAddr, updateOpinion)
-				if err != nil {
-					log.Fatal(err)
-				}
-				chainObserverBuf := &rollup.ChainObserverBuf{}
-				if err := proto.Unmarshal(chainObserverBytes, chainObserverBuf); err != nil {
-					log.Fatal(err)
-				}
-				chain, err = chainObserverBuf.UnmarshalFromCheckpoint(runCtx, restoreCtx, checkpointer)
+				err := checkpointer.RestoreLatestState(
+					runCtx,
+					clnt,
+					rollupAddr,
+					updateOpinion,
+					func(chainObserverBytes []byte, restoreCtx checkpointing.RestoreContext) {
+						chainObserverBuf := &rollup.ChainObserverBuf{}
+						if err := proto.Unmarshal(chainObserverBytes, chainObserverBuf); err != nil {
+							log.Fatal(err)
+						}
+						var err error
+						chain, err = chainObserverBuf.UnmarshalFromCheckpoint(runCtx, restoreCtx, checkpointer)
```

Comment:
Just to be clear, does this callback function run synchronously with the RestoreLatestState function? It should be that the callback has already finished running when the outer function returns for this code to be safe. That property should be clearly stated in the code since callbacks tend to be asynchronous rather than synchronous.

Reply:
Yes, it runs synchronously.  I changed the parameter name to "unmarshalFunc" (rather than "callback") and noted in a comment on the interface that the callback is synchronous.